pkg/virtual/apiexport/authorizer/binding.go

```diff
@@ -35,18 +37,25 @@ import (
 )
 
 type boundAPIAuthorizer struct {
+	getAPIExport          func(clusterName, apiExportName string) (*apisv1alpha2.APIExport, error)
 	getAPIBindingByExport func(clusterName, apiExportName, apiExportCluster string) (*apisv1alpha2.APIBinding, error)
 
 	delegate authorizer.Authorizer
 }
 
+const wildcardVerb = "*"
+
 var readOnlyVerbs = []string{"get", "list", "watch"}
 
-func NewBoundAPIAuthorizer(delegate authorizer.Authorizer, apiBindingInformer apisv1alpha2informers.APIBindingClusterInformer, kubeClusterClient kcpkubernetesclientset.ClusterInterface) authorizer.Authorizer {
+func NewBoundAPIAuthorizer(delegate authorizer.Authorizer, apiBindingInformer apisv1alpha2informers.APIBindingClusterInformer, apiExportInformer apisv1alpha2informers.APIExportClusterInformer, kubeClusterClient kcpkubernetesclientset.ClusterInterface) authorizer.Authorizer {
+	apiExportLister := apiExportInformer.Lister()
 	apiBindingLister := apiBindingInformer.Lister()
 
 	return &boundAPIAuthorizer{
 		delegate: delegate,
+		getAPIExport: func(clusterName, apiExportName string) (*apisv1alpha2.APIExport, error) {
+			return apiExportLister.Cluster(logicalcluster.Name(clusterName)).Get(apiExportName)
+		},
```

Comment:
Remind me, will this work if APIExport is on another shard? Is Lister here cache aware?